src/tt/client.h

```diff
@@ -10,7 +10,7 @@
 #ifndef IREE_PJRT_PLUGIN_PJRT_CPU_CLIENT_H_
 #define IREE_PJRT_PLUGIN_PJRT_CPU_CLIENT_H_
 
-#include "common/api_impl.h"
+#include "common/pjrt_implementation/client_instance.h"
```

Comment:
Thanks for doing this! Since we only have one client, does it make sense to eliminate the tt and common directories and just have everything live in pjrt_implementation?

Reply:
Thank you for suggestion Aleks, it makes sense for sure, we plan to change the folder structure and reorganize the code further, this PR is just one step in this process - to split the `api_impl.h/cc` into multiple files so we can read and understand the code easier. We will make further changes in some of the next PRs.